test/util/test_file_io.py

```diff
@@ -1,11 +1,61 @@
+from __future__ import annotations
+
 from pathlib import Path
+from unittest import TestCase
 from unittest.mock import MagicMock, patch
 
 from pytest import raises
 
 from flash_patcher.exception.error_manager import ErrorManager
 from flash_patcher.exception.injection import InjectionError
-from flash_patcher.util.file_io import read_safe, readlines_safe, writelines_safe
+from flash_patcher.util.file_io import FileWritebackManager, \
+    read_safe, readlines_safe, writelines_safe
+
+class FileWritebackManagerSpec (TestCase):
+    @patch('pathlib.Path.open', create=True)
+    def test_rw_safe_success(
+        self: FileWritebackManagerSpec,
+        mock_open: MagicMock,
+    ) -> None:
+        mock_file = MagicMock()
+
+        mock_open.return_value.__enter__.return_value = mock_file
+
+        with open("../test/testdata/DoAction1.as", encoding="utf-8") as file:
+            content_expected = file.read()
+            mock_file.read.return_value = content_expected
```

Comment:
Question: is it safe/reasonable to dedent this line? I've always tried to follow the principle of keeping a file open for as little time as possible, but I'm not sure if that's reasonable.

Would keeping this indented perhaps have safety benefits?

Reply:
Yes, moving this outside of the `with` block is fine. Will fix.